src/montecarlo.h

```diff
@@ -2,82 +2,121 @@
 #ifndef FAUNUS_MONTECARLO_H
 #define FAUNUS_MONTECARLO_H
 
-#include "energy.h"
-#include "move.h"
+#include "space.h"
+#include <memory>
 
 namespace Faunus {
-class MCSimulation {
-  private:
-    typedef typename Space::Tpvec Tpvec;
 
-    std::string lastMoveName; //!< name of latest move
+namespace Energy {
+class Hamiltonian;
+}
 
-    spdlog::level::level_enum log_level; //!< Storage for original loglevel
+namespace Move {
+class Movebase;
+class Propagator;
+} // namespace Move
 
-    bool metropolis(double du) const; //!< Metropolis criterion (true=accept)
+namespace MPI {
+class MPIController;
+}
 
+/**
+ * @brief Class to handle Monte Carlo moves
+ *
+ * This class maintains a list of MC moves and takes care of
+ * randomly selecting a moves, perform the move, decide if the move
+ * if accepted, and also checks if there is a drift in the potential
+ * energy.
+ *
+ * The moves always operate on a trial state (`new_state`). If accepted,
+ * the changes will be copied into the `old_state`. The state here refers
```

Comment:
Why not to name the state variables more descriptively, e.g., `trial_state` and `default_state` or similarly?

Reply:
indeed, `trial_state` and merely `state`?